### ISSUE TITLE
Parsing error: Type expected with new eslint projectService option

### ISSUE THREAD

**Issue by falco467** [ota-meshi/astro-eslint-parser#331]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I'm using [eslint-plugin-astro](https://github.com/ota-meshi/eslint-plugin-astro). (`*.astro` file linting does not work with the parser alone. You should also use eslint-plugin-astro with it.)
- [X] I'm sure the problem is a parser problem. (If you are not sure, search for the issue in [eslint-plugin-astro](https://github.com/ota-meshi/eslint-plugin-astro) repo and open the issue in [eslint-plugin-astro](https://github.com/ota-meshi/eslint-plugin-astro) repo if there is no solution.
- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.

### What version of ESLint are you using?

9.6.0

### What version of `eslint-plugin-astro` and `astro-eslint-parser` are you using?

- astro-eslint-parser@1.0.2
- eslint-plugin-astro@1.2.3

### What did you do?

<details>
<summary>Configuration</summary>

```
import eslint from '@eslint/js'
import globals from 'globals'
import tsEslint from 'typescript-eslint'
import epAstro from 'eslint-plugin-astro'
import astroParser from 'astro-eslint-parser'

export default [
  eslint.configs.recommended,
  ...tsEslint.configs.strictTypeChecked,
  ...tsEslint.configs.stylisticTypeChecked,
  ...epAstro.configs.recommended,
  {
    languageOptions: {
      parser: tsEslint.parser,
      parserOptions: {
        projectService: true,
        extraFileExtensions: ['.astro'],
      },
      globals: {
        ...globals.browser
      }
    },
  },
  {
    files: ['**/*.astro'],
    languageOptions: {
      parser: astroParser,
      parserOptions: {
        parser: tsEslint.parser,
      }
    },
  },
]

```
</details>

```astro
---
import BasicPage from '.../BasicPage.astro'
import Welcome from '../welcome.svelte'
---

<BasicPage title="Title">
  <Welcome client:load/>
</BasicPage>
```

### What did you expect to happen?

No errors

### What actually happened?

6:1  error  Parsing error: Type expected

### Link to Minimal Reproducible Example

The errors appeared as soon as I changed eslint config `project: true` to the new recommended `projectService: true`.

**Comment by shepherdjerred** [comment]
I've experienced this as well. Changing `projectService` back to `project` fixes the issue.

**Comment by JoshuaKGoldberg** [comment]
Could someone please post a full reproduction that I can run to try to help with this? Just a few snippets that don't include the full `package.json` `"dependencies"`+`"devDependencies"` versions aren't enough to go off of.

**Comment by falco467** [comment]
@shepherdjerred can you provide a reproducible example? I've recently changed too much in our project to get it to reproduce.

**Comment by shepherdjerred** [comment]
This occurred on commit `7432d8e` of my [personal website](https://github.com/shepherdjerred/sjer.red).

Steps:

1. Run `npm i`
1. Change project -> projectService to `true` in `eslint.config.mjs`
1. Run npx eslint

Output: https://gist.github.com/shepherdjerred/21bd162b61c107fefc405a07fb6ae3ee

**Comment by shepherdjerred** [comment]
Here's a reproduction using my [personal website](https://github.com/shepherdjerred/sjer.red).

```bash
git clone https://github.com/shepherdjerred/sjer.red
# this branch has projectService set to true
git checkout sj/project-service
npm ci
# to generate the .astro folder w/ types
npm run build
npx eslint
```

You'll see the following: https://gist.github.com/shepherdjerred/465d42188031dcd7ea7a1c4db9d7043e

**Comment by JoshuaKGoldberg** [comment]
tl;dr: use `files: ['**/.astro']`; right now the Astro parser isn't in play

---

Hmm, from https://github.com/shepherdjerred/sjer.red/blob/5d7d9fc6a1d767e46bd84c6b9cce8d91fa1d57a1/eslint.config.mjs#L24-L32:

```plaintext
{
  files: ["*.astro"],
  parser: "astro-eslint-parser",
  parserOptions: {
    parser: "@typescript-eslint/parser",
    extraFileExtensions: [".astro"],
  },
  rules: {},
},
```

Two problems there:

* It should be `**/*.astro` to match files in nested directories, not just the root directory.
* `parser` in ESLint flat config is specified as the parser object itself, not a string path.

When I change to `files: ["**/*.astro"]` I get this error:

```plaintext
$ npx eslint                                                                

Oops! Something went wrong! :(

ESLint: 9.9.0

A config object is using the "parser" key, which is not supported in flat config system.

Flat config uses "languageOptions.parser" to override the default parser.

Please see the following page for information on how to convert your config object into the correct format:
https://eslint.org/docs/latest/use/configure/migration-guide#custom-parsers

If you're not using "parser" directly (it may be coming from a plugin), please see the following:
https://eslint.org/docs/latest/use/configure/migration-guide#using-eslintrc-configs-in-flat-config
```

https://github.com/JoshuaKGoldberg/repros/tree/tseslint-on-astro shows a minimum reproduction of running the typescript-eslint parser on an `.astro` file. Voila, same error!

I think the main takeaway here is that the error message should ideally contain the parser... which, what do you know, I had filed on ESLint back in 2022! https://github.com/eslint/eslint/issues/16565

The ESLint TSC has in the past generally pretty unwilling to reconsider closed issues after the fact. You could try filing a new one, and I'd voice my opinion in support (again). But 🤷 no guarantees.

**Comment by falco467** [comment]
@JoshuaKGoldberg 
Here is a minimal reproducible example:

eslint.config.js
```javascript
import epAstro from 'eslint-plugin-astro'
import astroParser from 'astro-eslint-parser'

export default [
  ...epAstro.configs.recommended,
  {
    files: ['**/*.astro'],
    languageOptions: {
      parser: astroParser,
      parserOptions: {
        projectService: true,
      }
    },
  },
]
```

Parsing breaks as soon as projectService is set to true (which is needed to work with typescript) - and with --print-config we can also see that astro-eslint-parser is configured as the parser in the mix, but the error is thrown by typescript-eslint. It seems it will forward some code to the typescript-eslint-parser, which it cannot read as valid javascript.

As a workaround I have now disabled all type-aware rules for astro files like this:
```
export default [
...[
    ...tsEslint.configs.strictTypeChecked,
    ...tsEslint.configs.stylisticTypeChecked,
    love
  ].map(config => ({...config, ignores:["**/*.astro"]})),
  {
    files: ['**/*.astro'],
    languageOptions: {
      parser: astroParser,
      parserOptions: {
        projectService: false,
        parser: null
      }
    },
  },
```
This will disable the typescript parser for astro files and will just lint normal eslint errors. Sadly without any type-aware errors, since I cannot get astro-eslint-parser to work with typescript-eslint.

**Comment by azat-io** [comment]
The same problem.

`projectService: false` works fine for me.

**Comment by JoshuaKGoldberg** [comment]
Thanks for the file texts! I put up a [full minimum repro](https://antfu.me/posts/why-reproductions-are-required#reproducible-projects-or-playgrounds) here: https://github.com/JoshuaKGoldberg/repros/tree/astro-eslint-parser-tseslint-projectservice

@ota-meshi I think this is indeed an issue in how astro-eslint-parser is calling to the tseslint parser. Could this be reopened please?

**Comment by jasikpark** [comment]
Just ran into this, sorry for commenting on such a dead issue, but is there a fix for this?
